File: lib/common/models/ios/pose_estimation_settings.dart
class IOSPoseEstimationSettings {
  // Info messages
  String? faceIsOk;
  String? notInArea;
  String? faceTooSmall;
  String? faceTooBig;
  String? completed;
  String? turnRight;
  String? turnLeft;
  String? turnUp;
  String? turnDown;
  String? lookStraight;
  String? errorMessage;
  String? tryAgain;
  String? errorTitle;
  String? confirm;
  String? next;
  String? holdPhoneVertically;
  String? informationScreenDesc1;
  String? informationScreenDesc2;
  String? informationScreenTitle;
  String? wrongPose;
  String? descriptionHeader;
  // screen config
  String? appBackgroundColor;
  String? appFontColor;
  String? primaryButtonBackgroundColor;
  String? primaryButtonTextColor;
  String? ovalBorderColor;
  String? ovalBorderSuccessColor;
  String? poseCount;
  String? secondaryGuideVisibility;
  String? mainGuideVisibility;
  String? buttonRadious;
  int? manualCropTimeout;

  // main guide images
  String? mainGuideUp;
  String? mainGuideDown;
  String? mainGuideLeft;
  String? mainGuideRight;
  String? mainGuideStraight;

  // secondary guide images
  String? secondaryGuideUp;
  String? secondaryGuideDown;
  String? secondaryGuideLeft;
  String? secondaryGuideRight;

  IOSPoseEstimationSettings(
      {this.faceIsOk,
      this.notInArea,
      this.faceTooSmall,
      this.faceTooBig,
      this.completed,
      this.turnRight,
      this.turnLeft,
      this.turnUp,
      this.turnDown,
      this.lookStraight,
      this.errorMessage,
      this.tryAgain,
      this.errorTitle,
      this.confirm,
      this.next,
      this.holdPhoneVertically,
      this.informationScreenDesc1,
      this.informationScreenDesc2,
      this.informationScreenTitle,
      this.wrongPose,
      this.descriptionHeader,
      this.appBackgroundColor,
      this.appFontColor,
      this.primaryButtonBackgroundColor,
      this.primaryButtonTextColor,
      this.ovalBorderColor,
      this.ovalBorderSuccessColor,
      this.poseCount,
      this.mainGuideVisibility,
      this.secondaryGuideVisibility,
      this.buttonRadious,
      this.manualCropTimeout});

  Map toJson() {
    return {
      "faceIsOk": faceIsOk,
      "notInArea": notInArea,
      "faceTooSmall": faceTooSmall,
      "faceTooBig": faceTooBig,
      "completed": completed,
      "turnRight": turnRight,
      "turnLeft": turnLeft,
      "turnUp": turnUp,
      "turnDown": turnDown,
      "lookStraight": lookStraight,
      "errorMessage": errorMessage,
      "tryAgain": tryAgain,
      "errorTitle": errorTitle,
      "confirm": confirm,
      "next": next,
      "holdPhoneVertically": holdPhoneVertically,
      "informationScreenDesc1": informationScreenDesc1,
      "informationScreenDesc2": informationScreenDesc2,
      "informationScreenTitle": informationScreenTitle,
      "wrongPose": wrongPose,
      "descriptionHeader": descriptionHeader,
      "appBackgroundColor": appBackgroundColor,
      "appFontColor": appFontColor,
      "primaryButtonBackgroundColor": primaryButtonBackgroundColor,
      "primaryButtonTextColor": primaryButtonTextColor,
      "ovalBorderColor": ovalBorderColor,
      "ovalBorderSuccessColor": ovalBorderSuccessColor,
      "poseCount": poseCount,
      "mainGuideVisibility": mainGuideVisibility,
      "secondaryGuideVisibility": secondaryGuideVisibility,
      "buttonRadious": buttonRadious,
      "manualCropTimeout": manualCropTimeout
    };
  }
}

File: ios/Classes/models/PoseEstimationSettings.swift
//
//  PoseEstimationSettings.swift
//  flutter_amanisdk_v2
//
//  Created by Deniz Can on 14.11.2022.
//

struct PoseEstimationSettings: Codable {
  // info messages
  var faceIsOk: String = "Lütfen sabit durun."
  var notInArea: String = "Yüzünüzü daire içine hizalayın"
  var faceTooSmall: String = "Yüzünüz çok uzakta"
  var faceTooBig: String = "Yüzünüz çok yakında"
  var completed: String = "İşlem Tamam"
  var turnRight: String = "→"
  var turnLeft: String = "←"
  var turnUp: String = "↑"
  var turnDown: String = "↓"
  var lookStraight: String = "düz bak"
  var errorMessage: String = "Lütfen adımları dairenin içinden çıkmadan gerçekleştirin"
  var tryAgain: String = "Tekrar Dene"
  var errorTitle: String = "Başarısız"
  var confirm: String = "Onayla"
  var next: String = "Sonraki"
  var holdPhoneVertically: String = "Telefonu düz tutun"
  var informationScreenDesc1: String = "Doğrulamaya başlamak için, yüzünü alanın içinde tut"
  var informationScreenDesc2: String = ""
  var informationScreenTitle: String = "Selfie Doğrulama Talimatları"
  var wrongPose: String = "Yüzün düz konumda olmalı"
  var descriptionHeader: String = "Lütfen telefonu dik tutun ve yüzünü belirtilen alana hizalamaya özen göster"
  // screen config
  var appBackgroundColor: String = "000000"
  var appFontColor: String = "ffffff"
  var primaryButtonBackgroundColor: String = "ffffff"
  var primaryButtonTextColor: String = ""
  var ovalBorderColor: String = "ffffff"
  var ovalBorderSuccessColor: String = "00ff00"
  var poseCount: String = "3"
  var mainGuideVisibility: String = "true"
  var secondaryGuideVisibility: String = "false"
  var buttonRadious: String = "10"
  
  
  var mainGuideUp: String?
  var mainGuideDown: String?
  var mainGuideLeft: String?
  var mainGuideRight: String?
  var mainGuideStraight: String?
 
  var secondaryGuideUp: String?
  var secondaryGuideDown: String?
  var secondaryGuideLeft: String?
  var secondaryGuideRight: String?
  
  var manualCropTimeout: Int = 30
}
